test/test_service.rb

```diff
@@ -40,7 +40,7 @@ def test_construct_our_own_service
     assert_equal(created.spec.ports.size, our_service.spec.ports.size)
 
     # Check that original entity_config is not modified by kind/apiVersion patches:
-    assert_equal(our_service.kind, nil)
+    assert_nil our_service.kind
```

Comment:
Actually there's something related to that in the Ruby community style guide (https://github.com/bbatsov/ruby-style-guide#no-dsl-decorating):

> Omit both the outer braces and parentheses for methods that are part of an internal DSL.

With `assert_nil` being a DSL method I think it's a good point. Not trying to make you reconsider through.